tgui/packages/tgui/interfaces/Vote.js

```diff
@@ -0,0 +1,218 @@
+
+  return (
+    <Window
+      resizable
+      title={`Vote${
+        mode
+          ? `: ${
+            question
+              ? question.replace(/^\w/, c => c.toUpperCase())
+              : mode.replace(/^\w/, c => c.toUpperCase())
+          }`
+          : ""
+      }`}
+      width={400}
+      height={500} >
+      <Window.Content>
+        <Flex direction="column" height="100%">
+          {lower_admin && <AdminPanel />}
+          <ChoicesPanel />
+          <TimePanel />
+        </Flex>
+      </Window.Content>
+    </Window>
+  );
+};
+
+// Collapsible panel for admin actions.
+const AdminPanel = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { avm, avr, avmap, voting, upper_admin } = data;
+  return (
+    <Flex.Item>
+      <Collapsible title="Admin Options">
+        <Section mb={1} title="Start a vote">
+          <Flex justify="space-between">
+            <Flex.Item>
+              <Box mb={1}>
+                <Button
+                  disabled={!upper_admin || !avmap}
+                  onClick={() => act("map")}>
+                  Map
+                </Button>
+                {upper_admin && (
+                  <Button.Checkbox
+                    ml={1}
+                    color="red"
+                    checked={!avmap}
+                    onClick={() => act("toggle_map")}>
+                    Disable{!avmap ? "d" : ""}
+                  </Button.Checkbox>
+                )}
+              </Box>
+              <Box mb={1}>
+                <Button
+                  disabled={!upper_admin || !avr}
+                  onClick={() => act("restart")}>
+                  Restart
+                </Button>
+                {upper_admin && (
+                  <Button.Checkbox
+                    ml={1}
+                    color="red"
+                    checked={!avr}
+                    onClick={() => act("toggle_restart")}>
+                    Disable{!avr ? "d" : ""}
+                  </Button.Checkbox>
+                )}
+              </Box>
+              <Box mb={1}>
+                <Button
+                  disabled={!upper_admin || !avm}
+                  onClick={() => act("gamemode")}>
+                  Gamemode
+                </Button>
+                {upper_admin && (
+                  <Button.Checkbox
+                    ml={1}
+                    color="red"
+                    checked={!avm}
+                    onClick={() => act("toggle_gamemode")}>
+                    Disable{!avm ? "d" : ""}
+                  </Button.Checkbox>
+                )}
+              </Box>
+            </Flex.Item>
+            <Flex.Item>
+              <Button disabled={!upper_admin} onClick={() => act("custom")}>
+                Create Custom Vote
+              </Button>
+            </Flex.Item>
+          </Flex>
+        </Section>
+        <Section title="Still Voting">
+          <Collapsible title="View List">{voting}</Collapsible>
```

Comment:
Voting needs to be passed in as a list and handled with map and be contained in a locked size scrollable element.
Will probably look something like 
```
<Collapsible title="View List">
  <Box height="400px (or whatever)" overflowY="scroll">
  {voting.map(vote => (I can't remember how you actually format maps but its something like this)
    <Box (Table will probably be better than box since it has proper formatting for multiple rows. Pretty sure there is 1 for only 1 collumb, Grid maybe? idk)
       key={vote}>
       {vote}
    </Box>
   )}
  </Box>
</Collapsible>
```

Reply:
whatever works, but thats just an idea that might work based off what I remember from TGUI

Reply:
> Will probably look something like 
> 
> 
>   {voting.map(vote => (I can't remember how you actually format maps but its something like this)

Looks like you still remember it off the top of your head, that's correct